### PR TITLE
Adds test profile for qc workflow

### PR DESCRIPTION
**Pull request by Vlad-Dembrovskyi**
## Description
Adds a test profile to run qc sub-workflow of h3agwas pipeline with test data. 

For now can be only tested locally. Because as discussed [here](https://lifebit-biotech.slack.com/archives/C7NU05CP2/p1628680549031800), its not possible atm to set the mainScript with a command-line parameter which is required on CloudOS. How to do that will be tested with a separate PR.

## To test

```
nextflow run qc/main.nf -profile test
```
image

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -52,3 +52,28 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Nextflow
+        run: |
+          wget -qO- get.nextflow.io | bash
+          sudo mv nextflow /usr/local/bin/
+      - name: Pull docker image
+        run: |
+          docker pull quay.io/h3abionet_org/py3plink
+          docker pull quay.io/h3abionet_org/py3plink
```
Comment (dapineyro) [inline .github/workflows/ci.yml:73]: Does it need to be pulled twice?

File: qc/nextflow.config [hunk 2]
```diff
@@ -1,4 +1,35 @@
...
+    }
+  } else if(type == 'cpus'){
+    try {
+      return Math.min( obj, params.max_cpus as int )
+    } catch (all) {
+      println "   ### ERROR ###   Max cpus '${params.max_cpus}' is not valid! Using default value: $obj"
+      return obj
+    }
+  }
+}
```
Comment (dapineyro) [inline qc/nextflow.config:32]: Nice addition! :mechanical_arm:  I'll keep in mind in the future.

**Review by dapineyro — COMMENTED** [review]
Excellent work! Only few comments on `ci.yml`

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Vlad-Dembrovskyi) [reply .github/workflows/ci.yml:73]: good catch!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -52,3 +52,28 @@ jobs:
...
+          docker pull quay.io/h3abionet_org/h3agwas-texlive
+          docker pull quay.io/h3abionet_org/py2fastlmm
+          docker pull quay.io/h3abionet_org/py3metagwas
```
[the remark this answers is not in the document]
Comment (Vlad-Dembrovskyi) [reply .github/workflows/ci.yml:76]: The purpose of pulling the containers separately is to test if the pull step itself fails. But true that we don't need to do that in each step.

Will reduce the container pulling to only one of the tests

**Pull request merged**